packages/type-storage/src/utils/createFunction.ts

```diff
@@ -29,36 +29,37 @@ export function createFunction (
   prefix: Text | U8a,
   functionMetadata: StorageFunctionMetadata,
   options: CreateItemOptions = {}
-) {
-  let func: any; // The return value, it's a function that has metadata.
+): StorageFunction {
+  const meta = new StorageFunctionMetadata(functionMetadata);
+  const storageFn = meta as StorageFunction;
 
   if (options.isUnhashed) {
-    func = () => prefix.toU8a();
+    storageFn.create = () => prefix.toU8a();
```

Comment:
Sorry :( To get line 33 done, i.e. create a new instance from another one, had to do this. The assignment, i.e, Object.assign as removed doesn't actually do the trick...